auth/providers/azure/azure.go

```diff
@@ -120,23 +123,30 @@ type metadataJSON struct {
 func getMetadata(aadEndpoint, tenantID string) (*metadataJSON, error) {
 	metadataURL := aadEndpoint + tenantID + "/.well-known/openid-configuration"
 
-	response, err := http.Get(metadataURL)
+	request, err := http.NewRequest("GET", metadataURL, nil)
 	if err != nil {
 		return nil, err
 	}
-	defer response.Body.Close()
 
-	if response.StatusCode != http.StatusOK {
-		return nil, fmt.Errorf("get metadata url failed with status code: %d", response.StatusCode)
-	}
+	client := autorest.NewClientWithOptions(autorest.ClientOptions{})
+	client.Sender = httpclient.DefaultHTTPClient
+	client.PollingDuration = 30 * time.Second
```

Comment:
isn't Get Metadata a sync call? what is this polling duration trying to address?

Reply:
Polling duration is used as a per-request timeout:
https://pkg.go.dev/github.com/Azure/go-autorest/autorest#pkg-constants
https://pkg.go.dev/github.com/Azure/go-autorest/autorest#Client

Reply:
i'm not convinced. https://github.com/search?q=repo%3AAzure%2Fgo-autorest%20PollingDuration&type=code
I don't see any code referencing this.

Reply:
Looks like you're right, I'm too used to working with async operations. Updated to include a timeout in the default HTTP client, but I can move that to only affect this code if requested.